compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1208,8 +1202,9 @@ convertExpr env0 e = do
             let fldName = fldNames !! fldIndex
             recTyp <- convertType env (varType bind)
             pure $ mkDictProj env (fromTCon recTyp) fldName scrutinee' `ETmApp` EUnit
-    go env o@(Case scrutinee bind _ [alt@(DataAlt con, vs, x)]) args = fmap (, args) $ do
+    go env o@(Case scrutinee bind resultType [alt@(DataAlt con, vs, x)]) args = fmap (, args) $ do
         convertType env (varType bind) >>= \case
+            -- opaque types have no patterns that can be matched
```

Comment:
I assume that’s because in Core a pattern match on a literal is already an equality + a case on the result of that?

Reply:
Yes 🙃